pkg/userdata/coreos/provider.go

```diff
@@ -200,9 +201,13 @@ systemd:
         TimeoutStartSec=5min
         CPUAccounting=true
         MemoryAccounting=true
-        Environment=KUBELET_IMAGE=docker://k8s.gcr.io/hyperkube-amd64:{{ .HyperkubeImageTag }}
+{{- if .HTTPProxy }}
+        Environment=KUBELET_IMAGE=docker://{{ .HyperkubeImage }}:v{{ .KubeletVersion }}
+{{- else }}
+        Environment=KUBELET_IMAGE=docker://k8s.gcr.io/hyperkube-amd64:v{{ .KubeletVersion }}
+{{- end }}
         Environment="RKT_RUN_ARGS=--uuid-file-save=/var/cache/kubelet-pod.uuid \
-          --insecure-options=image \
+          --insecure-options=image{{if .InsecureHyperkubeImage }},http{{ end }} \
```

Comment:
Shouldn't the condition gate the whole arg and not only the args value?

Reply:
`image` is always required as the hyperkube images in `k8s.gcr.io` are not signed & signing is not default with docker.

I'm following the docker logic here, where `insecure` means http.